x-pack/plugins/alerting/server/rules_client/lib/siem_legacy_actions/migrate_legacy_actions.ts

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { AlertConsumers } from '@kbn/rule-data-utils';
+
+import type { SavedObjectReference } from '@kbn/core/server';
+import type { RulesClientContext } from '../..';
+import { RawRuleAction, RawRule } from '../../../types';
+import { validateActions } from '../validate_actions';
+import { injectReferencesIntoActions } from '../../common';
+import { retrieveMigratedLegacyActions } from './retrieve_migrated_legacy_actions';
+
+type MigrateLegacyActions = (
+  context: RulesClientContext,
+  params: {
+    ruleId: string;
+    references?: SavedObjectReference[];
+    actions?: RawRuleAction[];
+    attributes: RawRule;
+    skipActionsValidation?: boolean;
+  }
+) => Promise<{
+  resultedActions: RawRuleAction[];
+  resultedReferences: SavedObjectReference[];
+  hasLegacyActions: boolean;
+}>;
+
+/**
+ * @deprecated Once we are confident all rules relying on side-car actions SO's have been migrated to SO references we should remove this function
+ * migrates SIEM legacy actions and merges rule actions and references
+ * @param context RulesClient context
+ * @param params
+ * @returns
+ */
+export const migrateLegacyActions: MigrateLegacyActions = async (
```

Comment:
Can we add a try/catch here and log the error to not block the main method? what do you think?

Reply:
added